Feature/27: Modal 컴포넌트 작성.

Derek-94 opened the pull request:
### 구현내용

- modal 컴포넌트 작성
- modal storybook 제작.

Derek-94 committed:
feat(modal): Update Modal component, and storybook.

Derek-94 committed:
Add search template.

kdongha commented on components/molecules/Modal/index.tsx:
```diff
@@ -0,0 +1,55 @@
+    justify-contents: center;
+    overflow: auto;
+    margin-bottom: 1rem;
+    padding: ${(props) => (props.className === 'search' ? '0 40px' : '0 56px')};
```
```suggestion
    padding: 0 56px;
    &.search {
        padding: 0 40px;
    }
```

class name으로 하자면 이렇게 할수 있을꺼 같고
따로 props를 줘서 하는 방식도 좋을것 같습니다.

Derek-94 replied:
아 넵 감사합니다! 숙지해놓겠습니당.

kdongha reviewed (commented):
모달의 경우 ReactDOM. createPortal을 많이 쓰는데 react-modal의 경우도 createPortal을 쓰더라고요

modal용으로 따로 element를 만들어 주세요.

https://devrecipes.net/modal-component-with-next-js/

Derek-94 reviewed (commented):
네, 문서 읽어봤습니다.

_document.tsx 파일을 만들어 portal역할을 하는 `<div id = "modal-root"></div>` 를 만들고,

이를 `Modal` 컴포넌트에서 

```typescript
return ReactDOM.createPortal(
            <Modal>...</Modal>, 
            document.getElementById("modal-root")
        );
```

이렇게 해야할 것 같은데, 이런식으로 `components/molecules/Modal/index.tsx` 를 수정하면 storybook에서 렌더링을 하지 못하더라구요. 

위에 코드처럼 `createPortal` 사용하는 방법은 실제로 쓸때 사용하는 것 같은데,, 어떻게 생각하시는지 궁금합니다.

Derek-94 committed:
feat(feedback): Applied feedbacks.

- Add _document.tsx
- Make html file, add one div.
- Combined it with react-modal with setAppElement.

kdongha commented on .storybook/preview-head.html:
```diff
@@ -0,0 +1 @@
+<div id="modal-root"></div>
```
preview-head 파일이 아니라 preview-body 파일이 맞을꺼 같아요

Derek-94 replied:
아 넵 수정하고 바로 날리겠습니다.

Derek-94 committed:
refactor: Rename file.
